lms/djangoapps/certificates/api.py

```diff
@@ -463,51 +463,66 @@ def get_active_web_certificate(course, is_preview_mode=None):
     return None
 
 
-def get_certificate_template(course_key, mode, language):  # pylint: disable=unused-argument
+def get_certificate_template(course_key, mode, language):
     """
     Retrieves the custom certificate template based on course_key, mode, and language.
     """
-    org_id, template = None, None
+    template = None
     # fetch organization of the course
-    course_organization = get_course_organizations(course_key)
-    if course_organization:
-        org_id = course_organization[0]['id']
+    org_id = get_course_organization_id(course_key)
 
-    if org_id and mode:
-        template = CertificateTemplate.objects.filter(
+    # only consider active templates
+    active_templates = CertificateTemplate.objects.filter(is_active=True)
+
+    if org_id and mode:  # get template by org, mode, and key
+        org_mode_and_key_templates = active_templates.filter(
             organization_id=org_id,
-            course_key=course_key,
             mode=mode,
-            is_active=True
+            course_key=course_key
         )
-    # if don't template find by org and mode
-    if not template and org_id and mode:
-        template = CertificateTemplate.objects.filter(
+        template = get_templates_with_language_or_defaults(language, org_mode_and_key_templates)
+
+    # since no template matched that course_key, only consider templates with empty course_key
```

Comment:
How do we know no templates matched the course_key? The previous query was also checking for mode and organization_id. One of those may not have matched.

Reply:
Hmmmm this is the same logic as before (which a checked for a course key with org and mode as well), should this logic be updated to consider a template with the right course key an override, even if the org and mode don't match?

Reply:
I'm not sure, it might be worth asking Jasper for his thoughts. I would think we should consider the possibility that you may have a different template for different modes for a specific course, like a verified template and an honor code template, for example.

Reply:
Thats still possible with this logic, a template can be created with a specific mode in a specific course and will be returned when a certificate is being generated for a user enrolled in that course and mode.